### DISCUSSION TITLE
HTML format on save inconsistent & adding space for new lines

### DISCUSSION THREAD

**Discussion by ftr002** [zed-industries/zed#19109] [Feature Requests]
Hi there,

The HTML formatter is splitting some lines in weird ways and by doing so is adding a random extra space that is not consistent because the formatting is not consistent.

Does anyone know how to fix this?

```
                <ul>
                    <li>
                        This is a simple line of text that is being split onto a
                        separate line. It has an extra space at the start when
                        it is viewed in the browser.
                    </li>
                    <li>This stays on the same line</li>
                    <li>
                        This is another line of text that is being split onto a
                        separate line. Also has an extra space.
                    </li>
                    <li>This also stays on the same line.</li>
                </ul>
```

![image](https://github.com/user-attachments/assets/a6510d96-6e22-4e76-9574-9cfbccbc1ecb)

I'm not even sure why it is splitting some lines and not others even when it's on a full screen and there's plenty of room. Very strange. Not as big of an issue as the random space.

Please send help.

TY

**Comment by AlFirous** [comment]
I encounter same problem too. My current workaround is to disable the formatting on save.

`Settings -> Editor -> Formatting -> Format On Save -> Off`

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
